src/js/hooks/reporting/useExpirationHistoryReport.jsx

```diff
@@ -104,12 +111,37 @@ const useExpirationHistoryReport = ({
     serializedParams,
     setShouldFetch,
     filtersInitialized,
+    onFetchedData: (data) => {
+      setTableData((prevState) => ({
+        ...prevState,
+        totalValueLostToExpiry: data.totalValueLostToExpiry,
+        totalQuantityLostToExpiry: data.totalQuantityLostToExpiry,
+      }));
+    },
   });
 
   useEffect(() => {
     setTotalCount(tableData.totalCount);
   }, [tableData]);
 
+  // If the user switches to a different location, reset the table data.
+  // This prevents showing outdated data from the previous location
+  useEffect(() => {
+    // This if statement ensures that when the page loads for the first time,
+    // we don't trigger an unnecessary reset so this only runs when the location actually changes
+    if (previousLocation.current !== currentLocation?.id) {
+    // Reset table data to have clear data for the new location
+      setTableData({
+        data: [],
+        totalCount: 0,
+        totalValueLostToExpiry: 0,
+        totalQuantityLostToExpiry: 0,
+      });
+      // Update the reference to the current location for future checks
+      previousLocation.current = currentLocation?.id;
+    }
+  }, [currentLocation?.id]);
```

Comment:
this can potentially become a reusable hook that would accept a `callback` argument - in this case the `setTableData(...)` and in the other file, it would accept the `setFilterParams(..)`.
@alannadolny thoughts?